### PR TITLE
Fix JsException passing from pyodide environment to host

### PR DESCRIPTION
**Pull request by koenvo**
When a JsException exception is raised the pyodide environment pickles the exception and send it back to the host. The host tries to unpickle it and fails. This happens because `pyodide.JsException` does not exist.

This PR solves this issue by interception during unpickling.

note: maybe a more generic way of passing exceptions from pyodide environment to host is better? cloudpickle might do the trick.

### CODE REVIEW

**Review by hoodmane — APPROVED** [review]
Thanks @koenvo! It looks good to me.

**Comment by koenvo** [comment]
> Thanks @koenvo! It looks good to me.

I noticed some tests were failing. Hope all run correctly now.

File: tests/test_decorator.py [hunk 1]
```diff
@@ -69,6 +70,20 @@ def inner_function(selenium, x):
     assert inner_function(selenium, 6) == 7
 
 
+def test_inner_function_js_exception(selenium):
+    @run_in_pyodide
+    def inner_function(selenium):
+        from js import eval as js_eval
+
+        js_eval("throw 'some error'")
```
Comment (hoodmane) [inline tests/test_decorator.py:78]:
```suggestion
        from pyodide.code import run_js

        run_js("throw 'some error'")
```

**Review by rth — APPROVED** [review]
Thanks @koenvo . LGTM as well.

> note: maybe a more generic way of passing exceptions from pyodide environment to host is better? cloudpickle might do the trick.

Indeed that's a good question. We could consider switching to cloudpickle if we run into other frequent pickling issues.

**Comment by rth** [comment]
Could you please just add a changelog.md entry?

**Comment by ryanking13** [comment]
> > main / test (macos-latest, 0.21.0, selenium, safari, 1) (pull_request) Failing after
> 
> Not sure what's going on with that test, but it's failing quite frequently (unrelated to this PR).

It seems like safaridriver is a bit unstable. Probably we can disable a test using `selenium_standalone_refresh` fixture which fails more frequently.

**Pull request merged**